### PR TITLE
The first working prototype. It fetches titles.

### PR DESCRIPTION
**Pull request by Rembane**
* It connects to an IRC server, reads every line and picks out all
    URLs from the traffic. Then the bot fetches the pages associated
    with the URLs and gets the title from each page which it posts back
    to the channel it received it in.
  * There are tests!
  * The async library was used so it should be fairly stable.

### CODE REVIEW

File: src/Lib.hs [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+  , port     :: PortNumber
+  , chan     :: Text
+  , nick     :: Text
+  , password :: Maybe Text
+  , longname :: Text
+  }
+
+emptyConfig = Config
+  { server   = ""
+  , port     = 1024
```
Comment (Jassob) [inline src/Lib.hs]: FYI, IANA has reserved the port 1024. Not sure if this could pose a problem..

File: src/Lib.hs [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+  = ReplyWelcome B8.ByteString
+  | Ping B8.ByteString
+  | Privmsg B8.ByteString B8.ByteString B8.ByteString -- ^ Sender, Target, message
+  | Other B8.ByteString
+  deriving (Eq, Show)
+
+-- | Write a ByteString to the connection.
+writeRaw :: Connection -> B8.ByteString -> IO ()
+writeRaw connection t =
+  let t' = t <> "\r\n" in connectionPut connection t' *> B8.putStr t'
```
Comment (Jassob) [inline src/Lib.hs]: I'd probably written it like this (of course it's alright to keep the let-clause):

```suggestion
  connectionPut connection t *> B8.putStr $ t <> endOfLine
    where endOfLine = "\r\n"
```

This way it's more clear why we're appending "\r\n" to the bytestring.

**Review by Jassob — APPROVED** [review]
Looks good! Just a few comments, feel free to ignore them.

File: src/Lib.hs (hunk 2 above)
Comment (Rembane) [reply src/Lib.hs]: Yeah, but I would probably rather add a comment to the function that all IRC-commands are ended with \r\n. I'm not much for indirection.

File: src/Lib.hs (hunk 1 above)
Comment (Rembane) [reply src/Lib.hs]: That's a very good point, thank you for catching that one. I blame off by one error. Fix is on its way.

**Pull request merged**